app/views/find/v2/results/index.html.erb

```diff
@@ -18,9 +18,15 @@
           <%= form.govuk_check_box :can_sponsor_visa, "true", multiple: false, label: { text: t("helpers.label.courses_query_filters.can_sponsor_visa"), size: "s" } %>
         <% end %>
 
-        <%= form.govuk_check_boxes_fieldset :subjects, legend: { text: t("helpers.legend.courses_query_filters.secondary_html"), size: "s" }, hint: { text: t("helpers.hint.courses_query_filters.secondary"), class: "govuk-!-font-size-16" }, class: "app-filter__group", hidden: false, multiple: false, form_group: { class: "app-filter__group" } do %>
-          <%= form.govuk_collection_check_boxes :subjects, form.object.secondary_subjects, :subject_code, :subject_name, legend: nil, include_hidden: false, small: true %>
-        <% end %>
+        <div class="filter-search govuk-form-group app-filter__group" data-controller="filter-search">
+          <%= form.govuk_check_boxes_fieldset :subjects, legend: { text: t("helpers.legend.courses_query_filters.secondary_html"), size: "s" }, hint: { text: t("helpers.hint.courses_query_filters.secondary"), class: "govuk-!-font-size-16" }, class: "app-filter__group govuk-checkboxes--small", hidden: false, multiple: false do %>
+            <% form.object.secondary_subjects.each do |subject| %>
+              <%= form.govuk_check_box :subjects,
+              subject.subject_code,
+              label: { text: subject.subject_name } %>
+            <% end %>
```

Comment:
The govuk_collection_check_boxes with fieldset creates two classses `govuk-checkboxes` and  create more divs and form groups which makes the JS not work as expected. (i would have to call children many times. So I changed to loop through the secondary subjets.